[DSM] DDP-7174: DSM_use_role_migration

pegahtah opened the pull request:
DDP-7174

pegahtah committed:
DDP-7174
* Changed UserUtil.java and any class that was using it
* Created two Jdbi classes, `JdbiUser.java` and `JdbiUserRole.java` for new Database handling
* Changed `AuthenticationRoute.java` to use the guid from the db for the claims

pegahtah committed:
added todo

pegahtah committed:
Merge remote-tracking branch 'origin/DDP-7172-LiquiBase_User_Table' into DDP-7174

pegahtah committed:
creating AssigneeDto and removing Assignee.java

pegahtah committed:
Merge remote-tracking branch 'origin/DDP-7172-LiquiBase_User_Table' into DDP-7174

pegahtah committed:
Merge remote-tracking branch 'origin/DDP-7172-LiquiBase_User_Table' into DDP-7174

# Conflicts:
#	pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/DashboardRoute.java

pegahtah committed:
participant list loading working with new db

pegahtah committed:
Added Daos and Dtos and Jdbi classes for User, UserRoleDao, UserPermissions and UserSettingsDao

Changed other classes to use the shared db

pegahtah committed:
Merge remote-tracking branch 'origin/DDP-7172-LiquiBase_User_Table' into DDP-7174

# Conflicts:
#	pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/DashboardRoute.java

pegahtah committed:
added code for migrating roles and users

pegahtah committed:
inserting auth0 user id when missing
changing UserSettingsDao
changing user id from int to long
adding study_guid to ddp_instance queries

pegahtah committed:
Merge branch 'DDP-7172-LiquiBase_User_Table' into DSM_use_role_migration

pegahtah committed:
Merge remote-tracking branch 'origin/DDP-7172-LiquiBase_User_Table' into DSM_use_role_migration

gmakharat commented on pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/util/DdpDBUtils.java:
```diff
@@ -0,0 +1,58 @@
+
+        while (!unique) {
+            candidate = generator.get();
+            try (PreparedStatement stmt = handle.getConnection().prepareStatement(query)) {
```
Does not DSS already have class which is responsible to make queries to the database to check whether guid is unique or not?

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/roles/UserRoleDao.java:
```diff
@@ -0,0 +1,124 @@
+        return (List<String>) results.resultValue;
+    }
+
+    public static HashMap<Long, AssigneeDto> getAssigneeMap(String realm) {
```
why do we have this method as `static`?

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/roles/UserRoleDao.java:
```diff
@@ -0,0 +1,124 @@
+        return assignees;
+    }
+
+    public static Collection<AssigneeDto> getAssignees(String realm) {
```
`static` here as well

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/user/UserDao.java:
```diff
@@ -65,4 +69,83 @@ public int delete(int id) {
+            SimpleResult dbVals = new SimpleResult();
+            List<UserDto> res = handle.attach(JdbiUser.class).getUserMap();
+            Map<Long, String> map = new HashMap<>();
+            if (res != null) {
```
as far as I know if method of jdbi has `List` as return value and there are no results from `SQL` query it returns an empty list, maybe @Trimax can help us to clarify that, if so, we can ignore check on null.

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/user/UserDto.java:
```diff
@@ -3,28 +3,37 @@
+    private long userId;
+    private String guid;
+    private String firstName;
+    private String lastName;
```
what's the point of having `name` if we have `firstName` and `lastName`?

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/AuthenticationRoute.java:
```diff
@@ -54,24 +54,27 @@ public class AuthenticationRoute implements Route {
         this.auth0ClientId = auth0ClientId;
         this.auth0MgmntAudience = auth0MgmntAudience;
         this.audienceNameSpace = audienceNameSpace;
+        this.userDao = new UserDao();
```
if we pass `UserSettingsDao` through the constructor let's pass `UserDao` through the constructor to make this class fully dynamic.

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/AuthenticationRoute.java:
```diff
@@ -89,24 +92,25 @@ public Object handle(Request request, Response response) {
                         logger.info("User (" + email + ") was found ");
                         Gson gson = new Gson();
```
there is a singleton class called `ObjectMapperSingleton` which is wrapper of `jackson`, let's use it everywhere instead of `gson`

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/AuthenticationRoute.java:
```diff
@@ -121,7 +125,7 @@ public Object handle(Request request, Response response) {
                         }
                     } else {
-                        haltWithErrorMsg(400, response, "user was null");
+                        haltWithErrorMsg(400, response, "user info in token was null");
```
this class will rot, we can survive if we change nested if to guard conditions like:
`if (StringUtils.isBlank(auth0Token)) { haltWithErrorMsg(400, response, "user info in token was null"); }`
such structure to avoid nested if statements

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/UserSettingRoute.java:
```diff
@@ -27,8 +34,8 @@ public Object processRequest(Request request, Response response, String userId)
                     String requestBody = request.body();
-                    UserSettings userSettings = new Gson().fromJson(requestBody, UserSettings.class);
-                    UserSettings.editUserSettings(userIdRequest, userSettings);
+                    UserSettingsDto userSettingsDto = new Gson().fromJson(requestBody, UserSettingsDto.class);
```
let's use `ObjectMapperSingleton` here instead of `gson`

pegahtah committed:
Merge remote-tracking branch 'origin/DDP-7172-LiquiBase_User_Table' into DSM_use_role_migration

# Conflicts:
#	pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java
#	pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/DDPInstance.java
#	pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/ElasticSearchTest.java

pegahtah committed:
Merging develop

pegahtah committed:
checkstyle fixes

pegahtah committed:
checkstyle fixes

pegahtah committed:
Merge remote-tracking branch 'origin/DDP-7172-LiquiBase_User_Table' into DSM_use_role_migration

# Conflicts:
#	pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/util/LiquibaseUtil.java

pegahtah committed:
[DSM] DDP-7234: user_changes (#1612)

* DDP-7234 adding users

* DDP-7234 adding user_add permission
DDP-7234 adding new_user permissions

* DDP-7546 `GetRoleRoute` and relevant classes added

* changes to Dtos for json names

* DDP-7235 user access permission

* DDP-7235 assigning new roles to user

* DDP-7238 liquibase

* DDP-7234 changes to match the workflow of thee design

* DDP-7235
*removed new user role insertion query in `ddp-database`
*changed the `modify` functions to assign roles or change `user_profile` info

* DDP-7238
* adding `user_delete` permissions
* adding `DeactivateUserRoute.java` for deletion of user
* changed `User` and `Role` related classes to check for user being active
DDP-7235
*changes to `ModifyUserRoute` class to work with the designs

* DP-7234 Adding `study_admin` liquibase and instance role and role for studies that have phone numbers

* Merging develop

* checkstyle fixes

* checkstyle fixes

* DDP-7234 PR comments to use more Lombok annotations applied

* [DSM] DDP-7883_copy_dsm_user_id (#1641)

* DP-7238 Assignee table changes

* DP-7238 KitType changes to not use old access tables

* DP-7238 `ParticipantExit` changes to not use old access tables and get user names after retrieving exited pts

* DP-7238

* `SurveyTrigger` changes to not use old access tables and get user names after retrieving surveys
* Creating new method to return list of DSM users whether active or not

* DP-7238

* `AbstractionActivity` changes to not use old access tables and get user names after getting abstractions

* DP-7238

* `selectAllowedRealmsStartsLike` never used

* DP-7238

* change `KitDiscard` class for setting the usernames and also the `exitedKits` query in `application.conf`

* DP-7238

* change `ParticipantEvent` class for setting the usernames and also the `selectParticipantEvents` query in `application.conf`

* DDP-7883 removing queries from application.conf to classes

* checkstyle fixes

* DDP-7883 changing order to work with AllArgsConstructor
